full-service/src/db/txo.rs

```diff
@@ -1815,6 +1822,79 @@ mod tests {
         assert!(bob_txo.key_image.is_some());
     }
 
+    #[test_with_logger]
+    fn test_list_spendable_with_large_values(logger: Logger) {
+        let mut rng: StdRng = SeedableRng::from_seed([20u8; 32]);
+
+        let db_test_context = WalletDbTestContext::default();
+        let wallet_db = db_test_context.get_db_instance(logger);
+        let conn = wallet_db.get_conn().unwrap();
+
+        let root_id = RootIdentity::from_random(&mut rng);
+        let account_key = AccountKey::from(&root_id);
+        let (account_id_hex, _public_address_b58) = Account::create_from_root_entropy(
+            &root_id.root_entropy,
+            Some(1),
+            None,
+            None,
+            "Alice's Main Account",
+            "".to_string(),
+            "".to_string(),
+            "".to_string(),
+            &conn,
+        )
+        .unwrap();
+
+        // Create some very large TXOs for the account, above i64::MAX
+        let (_txo_hex, _txo, _key_image) = create_test_received_txo(
+            &account_key,
+            0,
+            Amount::new(u64::MAX, Mob::ID),
+            145,
+            &mut rng,
+            &wallet_db,
+        );
+
+        let (_txo_hex, _txo, _key_image) = create_test_received_txo(
+            &account_key,
+            0,
+            Amount::new(u64::MAX, Mob::ID),
+            146,
+            &mut rng,
+            &wallet_db,
+        );
+
+        // Create some small TXOs for the account
+        // [100, 200, 300, ... 2000]
+        for i in 1..20 {
+            let (_txo_hex, _txo, _key_image) = create_test_received_txo(
+                &account_key,
+                0,
+                Amount::new(100, Mob::ID), // 100.0 MOB * i
+                (146 + i) as u64,
+                &mut rng,
+                &wallet_db,
+            );
+        }
+
+        let spendable_txos = Txo::list_spendable(
+            Some(&account_id_hex.to_string()),
+            None,
+            None,
+            0,
+            Mob::MINIMUM_FEE,
+            &conn,
+        )
+        .unwrap();
+
+        // Max spendable SHOULD be u64::MAX * 2 + (100 * 14) - Mob::MINIMUM_FEE =
```

Comment:
If the loop above created txos in the range 100-2000 then the max should be u64::max * 2 + 21000 - fee